### ISSUE TITLE
App crashes in iOS with EXC_BAD_ACCESS exception

### ISSUE THREAD

**Issue by ameltom** [OneSignal/OneSignal-Unity-SDK#157]
**Description:**
App sometimes crashes after playing a few sessions, leaving the app on background in between.
The exception is EXC_BAD_ACCESS.
There is a release note on your Unity SDK v2.7.5 says you fix it (issue #391) but, It still happens on Unity SDK v2.7.6

**Environment**
Unity OneSignal SDK version 2.7.6
The issue occurs only in iOS

**Comment by jkasten2** [comment]
@ameltom Can you provide a full crash log from your device from XCode? We will need a stack trace to  look into this. Thanks!

**Comment by Nightsd01** [comment]
Closing due to no response

**Comment by rmgomez** [comment]
We are having the same issue. Here is the full stack trace of the crash.

**Environment**
Unity OneSignal SDK version 2.7.8
Unity v2018.2.20f1
The issue occurs only in iOS



[EXC_BAD_ACCESS-KERN_INVALID_ADDRESS.txt](https://github.com/OneSignal/OneSignal-Unity-SDK/files/2784439/EXC_BAD_ACCESS-KERN_INVALID_ADDRESS.txt)

**Comment by Nightsd01** [comment]
Thanks for posin the full stacktrace @rmgomez that is very helpful. Have you been able to reproduce it yourself or are you just seeing reports from end users?

Fortunately it looks like this crash is happening as iOS is terminating the application, so the app was already being terminated. However I will definitely investigate. If possible, the more detail you can provide the better!

**Comment by jkasten2** [comment]
Relevant part of the stack from the crash file that was attached.
```
#0. Crashed: com.apple.main-thread EXC_BAD_ACCESS KERN_INVALID_ADDRESS 0x00000000000001f8
0  game                           0x102a02b10 core::hash_set<core::pair<core::basic_string<char, core::StringStorageDefault<char> > const, int, true>, core::hash_pair<core::hash<core::basic_string<char, core::StringStorageDefault<char> > >, core::basic_string<char, core::StringStorageDefault<char> > const, int>, core::equal_pair<std::__1::equal_to<core::basic_string<char, core::StringStorageDefault<char> > >, core::basic_string<char, core::StringStorageDefault<char> > const, int> >::node* core::hash_set<core::pair<core::basic_string<char, core::StringStorageDefault<char> > const, int, true>, core::hash_pair<core::hash<core::basic_string<char, core::StringStorageDefault<char> > >, core::basic_string<char, core::StringStorageDefault<char> > const, int>, core::equal_pair<std::__1::equal_to<core::basic_string<char, core::StringStorageDefault<char> > >, core::basic_string<char, core::StringStorageDefault<char> > const, int> >::lookup<core::basic_string<char, core::StringStorageDefault<char> > >(core::basic_string<char, core::StringStorageDefault<char> > const&) const (hash_set.h:636)
1  game                           0x102a001e0 MonoManager::GetScriptingClass(char const*, char const*, char const*) (hash_set.h:308)
2  game                           0x10261580c InitializeCoreScriptingClasses() (CoreScriptingClasses.cpp:25)
3  game                           0x102615780 GetCoreScriptingClassesPtr() (CoreScriptingClasses.cpp:1686)
4  game                           0x1026155b4 GetCoreScriptingClasses() (CoreScriptingClasses.cpp:1695)
5  game                           0x102947994 NotifyPlayerQuit(bool) (Player.cpp:306)
6  game                           0x1029481d8 PlayerCleanup(bool) (Player.cpp:577)
7  game                           0x10276d11c UnityCleanup (LibEntryPoint.mm:220)
8  game                           0x1024ee528 -[UnityAppController applicationWillTerminate:] (UnityAppController.mm:410)
9  game                           0x103188de0 -[OneSignalAppDelegate oneSignalApplicationWillTerminate:] (UIApplicationDelegate+OneSignal.m:284)
```

@ameltom @rmgomez I reviewed the stack trace in the crash and although OneSignal is in the stack the crash is coming from `UnityAppController applicationWillTerminate`. The following Unity thread idicates it was fixed in Unity 2019.1.1f1
https://forum.unity.com/threads/crash-in-il2cpp-threadpool-worker-in-20-of-sessions-ios-only.542543/
